vms/platformvm/spend.go

```diff
@@ -254,6 +248,17 @@ func (vm *VM) stake(
 			})
 		}
 
+		// Handle fees after stake to reduce UTXO generation
```

Comment:
I'd move this change to dedicated PR, so we can separatly test and merge it without consideration of other changes